modules/wallet/update.go

```diff
@@ -0,0 +1,96 @@
+package wallet
+
+import (
+	"github.com/NebulousLabs/Sia/consensus"
+)
+
+// applyDiff will take the output and either add or delete it from the set of
+// outputs known to the wallet. If adding is true, then new outputs will be
+// added and expired outputs will be deleted. If adding is false, then new
+// outputs will be deleted and expired outputs will be added.
+func (w *Wallet) applyDiff(diff consensus.OutputDiff, adding bool) {
+	isNew := diff.New
+	if !adding {
+		isNew = !isNew
+	}
```

Comment:
just a thought, but maybe we should replace the `New`/`adding` boolean with an enum:
```go
const (
    DiffForward = iota
    DiffBackward
)
```